src/csvcubed/writers/qbwriter.py

```diff
@@ -820,6 +982,45 @@ def _define_csvw_column_for_qb_column(
             column.structural_definition.__class__.__name__,
         )
 
+        obs_val_col: Optional[QbColumn[QbObservationValue]] = None
+        obs_val_cols = get_columns_of_dsd_type(self.cube, QbObservationValue)
```

Comment:
Can you make sure you only populate these variables when inside the `if self.is_cube_in_pivoted_shape` block?